cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -179,9 +229,11 @@ private DatasetMeta getFromMds(Id.DatasetInstance instance) throws Exception {
    * @throws DatasetTypeNotFoundException if the dataset type was not found
    * @throws Exception if something went wrong
    */
-  public void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
+  void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
     // Throws NamespaceNotFoundException if the namespace does not exist
     Id.Namespace namespace = ConversionHelpers.toNamespaceId(namespaceId);
+    Principal principal = SecurityRequestContext.toPrincipal();
```

Comment:
Sounds good. I just don't have a good feeling about that logic in DatasetServiceClient. I think in the other branch I pointed to, we can inject an AuthenticationContext here, which will have an implementation that is either HTTP Based or UGI based will fix this. Is it ok if make the changes in that other branch?